backend/api/pdf_progress.py

```
@@ -0,0 +1,215 @@
+        if session_id in self.progress_data:
+            del self.progress_data[session_id]
+        logger.info(f"WebSocket disconnected for session: {session_id}")
+    
+    async def send_progress(self, session_id: str, progress_data: Dict[str, Any]):
+        """Send progress update to a specific session"""
+        if session_id in self.active_connections:
+            try:
+                websocket = self.active_connections[session_id]
+                await websocket.send_text(json.dumps(progress_data))
+                logger.debug(f"Sent progress update to {session_id}: {progress_data}")
+            except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)